zap/src/main/resources/org/zaproxy/zap/resources/xml/report.html.xsl

```diff
@@ -119,6 +122,42 @@ ZAP Scanning Report
   </tr>
 </table>
 <div class="spacer-lg"></div>
+
+<h3>Alerts</h3>
+<table width="75%" class="alerts">
+  <tr bgcolor="#666666"> 
+    <th width="60%" height="24">Name</th>
+    <th width="20%" align="center">Risk (Confidence)</th>
+    <th width="20%" align="center">Number of Instances</th>
+  </tr>
+  <xsl:key name="alerts-by-name-risk" match="alertitem" use="concat(name, ' ', riskdesc)"/>
+  <xsl:for-each select="descendant::alertitem[count(. | key('alerts-by-name-risk', concat(name, ' ', riskdesc))[1]) = 1]">
+    <xsl:sort order="descending" data-type="number" select="riskcode"/>
+    <xsl:sort order="descending" data-type="number" select="confidence"/>
+    <xsl:sort order="ascending" data-type="text" select="name"/>
+    <tr bgcolor="#e8e8e8"> 
+      <td>
+        <xsl:value-of select="name"/>
+      </td>
+      <td align="center">
+        <xsl:value-of select="riskdesc"/>
+      </td>
+      <td align="center">
+        <xsl:variable name="alerts-of-a-kind" select="key('alerts-by-name-risk', concat(name, ' ', riskdesc))"/>
+        <!-- Add <count>s when merge is on -->
+        <xsl:if test="$alerts-of-a-kind/count">
```

Comment:
- Could use choose to avoid repeating the (negated) condition
 - I'd suggest "name" (or "type") instead of "kind".